File: libs/jmap/src/main/java/rs/ltt/jmap/examples/entity/CreateMaskedEmailRequest.java
package rs.ltt.jmap.examples.entity;

import lombok.EqualsAndHashCode;
import lombok.NonNull;
import lombok.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import rs.ltt.jmap.common.entity.AbstractIdentifiableEntity;

@EqualsAndHashCode(callSuper = false)
@Value
public class CreateMaskedEmailRequest extends AbstractIdentifiableEntity {
  @NonNull
  String state;
  @NonNull
  String description;
  @Nullable
  String url;
  @NotNull
  String emailPrefix;
}

File: impl/fastmail/src/main/kotlin/com/restium/impl/fastmail/model/CreateMaskedEmailRequest.kt
package com.restium.impl.fastmail.model

import com.restium.jmap.entity.SetMaskedEmailRequest

typealias MaskedEmailRequestDto = SetMaskedEmailRequest

data class CreateMaskedEmailRequest(
  val description: String
) {
  private val state: String = "enabled"
  private val url: String? = null
  private val emailPrefix: String = ""

  fun toDto() = MaskedEmailRequestDto(state, description, url, emailPrefix)
}
